packages/dd-trace/src/llmobs/util.js

```diff
@@ -11,6 +13,162 @@ function encodeUnicode (str) {
   }).join('')
 }
 
+function validKind (kind) {
+  return SPAN_KINDS.includes(kind)
+}
+
+function getName (kind, options = {}, fn) {
+  return options.name || fn?.name || kind || ''
+}
+
+// extracts the argument names from a function string
+function parseArgumentNames (str) {
```

Comment:
This is parsing the argument and function signature from the entire serialized function? 💀

Reply:
yeah 💀 but i memoized it with a weak reference to the function, so if the same function is invoked a million times, it'll use the parsed argument names from the first time and map over the arguments accordingly. tbh this might need some iteration, because i tried to assume user cases in writing [tests](https://github.com/DataDog/dd-trace-js/blob/49c20e8d07c5767173a3e39088e2180cc1e62c49/packages/dd-trace/test/llmobs/util.spec.js#L55-L156) but i'm sure i didn't think of everything. `try/catch`ed it so we will never crash, but will probably iterate on it accordingly from user reports.